### PR TITLE
doc/dev: update logging conventions, align existing usage

### PR DESCRIPTION
**Pull request by bobheadxi**
Outlines some logging conventions based on our recent migration work, and update some existing callsites to align with the new recommendations

The main document to review is `doc/dev/how-to/add_logging.md`

Closes https://github.com/sourcegraph/sourcegraph/issues/36734

## Test plan

n/a, docs and minor logging usage changes but nothing major

### CODE REVIEW

File: doc/dev/how-to/add_logging.md [hunk 1]
```diff
@@ -46,150 +49,181 @@ func main() {
...
+Once initialized, you can use `log.Scoped()` to create some top-level loggers to propagate, from which you can:
 
-In a particular scope some fields might be repeatedly emitted. Consider creating a sub-logger in the scope by using `logger.With(...fields)`.
+- [create sub-loggers](#sub-loggers)
+- [write log entries](#writing-log-messages)
 
-```go
-func (s *Service) MyService(ctx context.Context, logger log.Logger) {
-    subLogger := logger.With(log.Int("id", s.ID), log.String("service", s.Name))
+The first top-level scoped logger is typically `"server"`, since most logging is related to server initialization and service-level logging - the name of the service itself is already logged as part of the [`Resource` field](https://opentelemetry.io/docs/reference/specification/logs/data-model/#field-resource) provided during initialization.
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:73]: FYI @burmudar since this goes against your previous recommendation

**Review by jhchabran — APPROVED** [review]

**Review by burmudar — APPROVED** [review]

**Review by keegancsmith — APPROVED** [review]
LGTM. Is cmd/searcher already using best practices? :D

**Comment by bobheadxi** [comment]
> Is cmd/searcher already using best practices? :D

I made some potential changes https://github.com/sourcegraph/sourcegraph/pull/36934/commits/81db47db7465800f969b2a8a0b64c631d5a0bfd6:

- `service` -> `server` - the struct is called `Service`, but this is just for the sake of consistency with other services
- Give `search.Store` its own scope

**Comment by bobheadxi** [comment]
Merging this for now, @keegancsmith let me know if you have any thoughts on the searcher scope changes, happy to follow up with additional changes

**Pull request merged**